src/SiteInformation/ContactAndLocation/EditProtocolsModal.js

```diff
@@ -0,0 +1,182 @@
+  DialogContent,
+  FormControl,
+  FormControlLabel,
+  Grid,
+  Select,
+  MenuItem,
+  Checkbox,
+  Button,
+  Snackbar,
+} from '@material-ui/core';
+import MuiAlert from '@material-ui/lab/Alert';
+import { useSelector, useDispatch } from 'react-redux';
+import { setEditProtocolsModalOpen } from '../../Store/actions';
+import { callAzureFunction } from '../../utils/SharedFunctions';
+import { useAuth0 } from '../../Auth/react-auth0-spa';
+
+// Helper function
+function Alert(props) {
+  return <MuiAlert elevation={6} variant="filled" {...props} />;
+}
+
+// A protocol dictionary in case labels are to be changed in the future
+const protocolDict = {
+  sensor_data: 'sensor_data',
+  bulk_density: 'bulk_density',
+  corn_disease: 'corn_disease',
+  farm_history: 'farm_history',
+  soil_texture: 'soil_texture',
+  gps_locations: 'gps_locations',
+  soil_nitrogen: 'soil_nitrogen',
+  yield_monitor: 'yield_monitor',
+  decomp_biomass: 'decomp_biomass',
+  cash_crop_yield: 'cash_crop_yield',
+  in_field_biomass: 'in_field_biomass',
+  weed_visual_rating: 'weed_visual_rating',
+  weed_quadrat_photos_beta: 'weed_quadrat_photos_beta',
+};
+
+const EditProtocolModal = () => {
+  const open = useSelector((state) => state.tableData.editProtocolsModalOpen);
+  const editProtocolsModalData = useSelector((state) => state.tableData.editProtocolsModalData);
+  const dispatch = useDispatch();
+  const { getTokenSilently } = useAuth0();
+  const [farmCode, setFarmCode] = useState();
+  const [backedProtocols, setBackedProtocols] = useState([]);
+  const [protocols, setProtocols] = useState([]);
+  const [maxWidth, setMaxWidth] = useState('xs');
+  const [snackbarData, setSnackbarData] = useState({
+    open: false,
+    text: '',
+    severity: 'success',
+  });
+
+  useEffect(() => {
+    if (editProtocolsModalData) {
+      setFarmCode(editProtocolsModalData.code);
+      const protocols = editProtocolsModalData.protocols;
+      setProtocols(protocols);
+      setBackedProtocols(protocols);
+    }
+  }, [editProtocolsModalData]);
+
+  const handleMaxWidthChange = (event) => {
+    setMaxWidth(event.target.value);
+  };
+
+  const handleCheckBoxClick = (protocol) => {
+    if (protocols[protocol] === 1) {
+      setProtocols({ ...protocols, [protocol]: 0 });
+    } else if (protocols[protocol] === 0 || protocols[protocol] === null) {
+      setProtocols({ ...protocols, [protocol]: 1 });
+    }
+  };
+
+  const handleEditProtocolsModalClose = () => {
+    dispatch(setEditProtocolsModalOpen(!open));
+  };
+
+  const handleCancelButton = () => {
+    setProtocols(backedProtocols);
+    handleEditProtocolsModalClose();
+  };
+
+  const updateProtocolEnrollment = () => {
+    const data = {
+      protocols,
+    };
+    callAzureFunction(
+      data,
+      `crowndb/protocol_enrollment/${farmCode}`,
+      'POST',
+      getTokenSilently,
+    ).then((res) => {
+      if (res.response && res.response.status === 201) {
+        setBackedProtocols(protocols);
```

Comment:
will you add a success Snackbar here?

Reply:
that says `Successfully updated protocol_enrollment`

Reply:
Yes, should the modal automatically close after the update and then show the success snack bar or should the modal remain open while showing the snackbar?

Reply:
close the modal on success and then show the snackbar